src/backend/metal/src/native.rs

```diff
@@ -226,7 +232,64 @@ pub enum DescriptorSetBinding {
 }
 
 #[derive(Debug)]
-pub enum Memory {
+pub struct Memory {
+    pub(crate) heap: MemoryHeap,
+    pub(crate) allocations: Arc<Mutex<MemoryAllocations>>,
+    pub(crate) mapping: Mutex<Option<MemoryMapping>>,
```

Comment:
A mutex is not strictly necessary here I think as `memory` in `vkMapMemory` and `vkUnmapMemory` should be externally synch following Vulkan.

Reply:
`UnsafeCell` then?

Reply:
Actually, this is not a perf concern. We can leave a `//TODO` comment and proceed.